apps/frontend/src/components/RequestModal/RequestModal.tsx

```diff
@@ -1,47 +1,87 @@
-import { Modal } from "react-bootstrap";
-import { Form } from 'react-router-dom';
+import { Modal, Spinner } from "react-bootstrap";
 import styles from './RequestModal.module.css';
+import { RequestType, ResponseWithUser } from "../../types";
+import requestServices from "../../services/requests";
+import { useState } from "react";
+import { useRevalidator } from "react-router";
+import ResponseBox from '../ResponseBox/ResponseBox'
 
 interface Props {
   show: boolean,
-  handleClose: () => void,
+  handleCloseModal: () => void,
+  request: RequestType,
+  updateRequestList: (requests: Array<RequestType>) => void,
 }
 
-export default function RequestModal({ show, handleClose }: Props) {
-  function handleSubmit() {
-    handleClose();
-  }
+export default function RequestModal({ show, handleCloseModal, request }: Props) {
+  const [loading, setIsLoading] = useState(false);
+  const revalidator = useRevalidator();
 
-  return (
-    <Modal show={show} onHide={handleClose} animation={true} backdrop="static" centered>
-      <Modal.Header closeButton>
-        <Modal.Title>Create a request</Modal.Title>
-      </Modal.Header>
-      <Modal.Body>
-        <Form method='post' className={styles.form} name='new-request-form'>
-          <label className={styles.label} htmlFor='title'>Title:</label>
-          <input className={styles.input} type='text' name='title' id='title' required minLength={4}></input>
+  async function handleCloseRequest() {
+    try {
+      setIsLoading(true);
+      await requestServices.closeRequest(String(request.id))
+      setIsLoading(false);
+      handleCloseModal();
+      revalidator.revalidate();
+      
+    } catch(e) {
+      setIsLoading(false);
+      console.log(e);
+    }
+  }
 
-          <label className={styles.label} htmlFor='content'>Content:</label>
-          <textarea className={styles.textarea} name='content' id='content' required></textarea>
+  const responses = request.responses.map((responseObj: ResponseWithUser)=> {
+    return (
+      <ResponseBox response={responseObj} key={responseObj.id}/>
+    ) 
+  })
+ 
+  /**
+  * Based on the status of a user (creator, or viewer) and the status of the request 
+  (OPEN or CLOSED) returns the corresponding JSX for the request modal.
+  */
+  function displayRequestActions(request: RequestType) {
+    let user = localStorage.getItem('user');
+    let username = user ? JSON.parse(user).username : null;
+    if (username === request.user.user_name && request.status === "OPEN") {
+      return (
+        <button className={`${styles.btn} ${styles.closeBtn}`} onClick={handleCloseRequest}>Close request</button>
+      )
+    } else if (!(username === request.user.user_name) && request.status === "OPEN") {
```

Comment:
could be `!==` but also, I think this first part of the condition is redundant

Reply:
Yes, you're right. It is redundant. Will fix!